allofplos/article_class.py

```diff
@@ -858,6 +858,25 @@ def url(self):
         """
         return self.get_url(plos_network=self.plos_network)
 
+    @property
+    def taxonomy(self):
+        """Taxonomy information
+        """
+        tag_path_elements = ('/',
+                             'article',
+                             'front',
+                             'article-meta',
+                             'article-categories')
+        e_list = self.get_element_xpath(tag_path_elements=tag_path_elements)
+        taxonomy_item = []
+        for subj in e_list[0].getchildren():
+            if subj.values()[0] == 'Discipline-v3':
+                subj_list = []
```

Comment:
I'd prefer this to be a list of tuples than a list of lists.
A tuple comprehension would probably be enough since you're only iterating over the 'subject' items.

It should be ordered, right?

Reply:
Oh and technically there is no such thing as a tuple comprehension (as it creates a generator), but if you wrap that in a tuple call, then you get a tuple comprehension as though it were a built-in feature.

```python
tuple(for e in subj.iter('subject'))
```

Reply:
+1 to tuples idea.
Because the full taxonomy paths are somewhat repetitive, what about storing it as a dict of `level_1_taxo_terms`, `level_2_taxo_terms`, etc?

Reply:
Yes, the order is relevant since it goes from a broad subject to a very specific one.  Will make a touple for each taxonomy.

Reply:
> Because the full taxonomy paths are somewhat repetitive, what about storing it as a dict of level_1_taxo_terms, level_2_taxo_terms, etc?

That seems like it should be a different PR… the functionality right now is simple, which is great. But there we would need to decide on a convention and we don't yet know what that should be.

But you might be right that this could be stored somewhat differently in a way that makes it more intuitive. Can either @sbassi or @eseiver explain the reasoning for why the taxonomy works the way it does? Is it supposed to be tree-ish?

Reply:
it is not a tree in the sense there are several branch that leads to the same small branchs.

![screenshot from 2018-02-12 16-40-37](https://user-images.githubusercontent.com/238517/36127748-cc224cee-1013-11e8-9a87-a7fe9f4d601e.png)

Reply:
Are they to be treated as the "same" "leaf"? or are they actually different "leaves" since they have the different descent hierarchies? I.e., is `Biomass (ecology)` to be treated as the same entity regardless?

Where do the semantics of this taxonomy scheme come from? 

This should be documented so that people can know where to look if they're confused.